src/pip/_internal/cli/parser.py

```diff
@@ -253,14 +258,20 @@ def error(self, msg):
         self.exit(UNKNOWN_ERROR, "{}\n".format(msg))
 
 
-def invalid_config_error_message(action, key, val):
+def invalid_store_bool_message(key, value):
+    # type: (str, str) -> str
+    """Returns a better error message when invalid configuration option
+    of action store_false or store_true is provided.
+    """
+    return ("{} is not a valid value for {} option, "
+            "please specify a boolean value like yes/no, "
+            "true/false or 1/0 instead.").format(value, key)
+
+
+def invalid_count_message(key, value):
+    # type: (str, str) -> str
     """Returns a better error message when invalid configuration option
-    is provided."""
-    if action in ('store_true', 'store_false'):
-        return ("{0} is not a valid value for {1} option, "
-                "please specify a boolean value like yes/no, "
-                "true/false or 1/0 instead.").format(val, key)
-
-    return ("{0} is not a valid value for {1} option, "
-            "please specify a numerical value like 1/0 "
-            "instead.").format(val, key)
+    of action count is provided.
+    """
+    return ("{} is not a valid value for {} option, "
+            "please specify a natural number instead.").format(value, key)
```

Comment:
The wolfram definition of natural number (https://mathworld.wolfram.com/NaturalNumber.html) have definitions which both include and exclude 0, which I believe is a valid value here. Perhaps we can say `non-negative integers` here (https://mathworld.wolfram.com/NonnegativeInteger.html)

Reply:
Thanks, TIL about such ambiguity.